[DOCS] Adds how DFA jobs work section (#754)
elastic/stack-docs: docs/en/stack/ml/df-analytics/ml-dfa-phases.asciidoc

[role="xpack"]
[[ml-dfa-phases]]
=== How a {dfanalytics-job} works

> This page provides high-level background information on how {dfanalytics-jobs} 

Reviewer: > This page...

In my opinion, this introductory text isn't required.  It's a short page and the title is quite clear, so I don't think people will be confused when they land here.

work, what kind of phases the jobs have, and what happens during the individual 
phases.

[discrete]

> ==== The phases of a {dfanalytics-job}

Reviewer: If you agree that the introductory sentence is unnecessary, I'd also suggest removing this section title.

> 

Reviewer: I think this sentence should be moved to the beginning of the paragraph:
```suggestion
A {dfanalytics-job} is essentially a persistent {es} task.
```

> During its life cycle, a {dfanalytics-job} goes through four phases regardless 

Reviewer: I don't think the "regardless..." is necessary, or if it is I think it should be a new sentence.
```suggestion
During its life cycle, a {dfanalytics-job} goes through four phases.
```

> of the {dfanalytics} type it performs. The phases are discrete steps of the 

Reviewer: If you agree with my other suggestions, this text is no longer required....
```suggestion

```

> {dfanalytics-job} that is essentially a persistent {es} task. The four phases 

Reviewer: Likewise:
```suggestion

```

> are the following:

Reviewer: Ditto
```suggestion

```

* reindexing,
* loading data,
* analyzing,
* writing results.

Let's take a look at the phases one-by-one.

[discrete]

> ===== Reindexing

Reviewer: ```suggestion
==== Reindexing
```

During the reindexing phase the documents from the source index or indices are 

> copied to the destination index. If the destination index does not exist, then 

Reviewer: Perhaps this is simpler?
```suggestion
copied to the destination index. If you want to define settings or mappings, you can create the index yourself. Otherwise,
```

> the job creates it before reindexing by using default settings. If it is already 

Reviewer: I think some of this can be simplified, since it seems like common sense
```suggestion
the job creates it using default settings.
```

> created, then the job will use the existing index. This provides flexibility for 

Reviewer: ```suggestion
This provides flexibility for 
```

> created, then the job will use the existing index. This provides flexibility for 

Reviewer: If you agree with my previous suggestions, these subsequent sentences are unnecessary:

> created.... want to have in it.

you to create the destination index and define any settings or mappings that you 
want to have in it. 

Once the destination index is built, the {dfanalytics-job} task calls the {es} 
{ref}/docs-reindex.html[Reindex API] to launch the reindexing task.

[discrete]

> ===== Loading data

Reviewer: ```suggestion
==== Loading data
```

After the reindexing is finished, the job fetches the needed data from the 

> destination index, converts it into the format that the analysis process 

Reviewer: Not required, but I like shorter sentences:
```suggestion
destination index. It converts the data into the format that the analysis process 
```

expects, then sends it to the analysis process.

[discrete]

> ===== Analyzing

Reviewer: ```suggestion
==== Analyzing
```

When all the required data is loaded, the analyzing phase begins. The exact 
process always depends on the type of {dfanalytics} – for example, 

> {classification} or {oldetection} – that is executed.

Reviewer: Can we provide more info here?  For example, something like "this is when the job starts generating models in xyz index" or some other juicy tidbit?

[role="xpack"]
[[ml-dfa-phases]]
=== How a {dfanalytics-job} works

This page provides high-level background information on how {dfanalytics-jobs} 
work, what kind of phases the jobs have, and what happens during the individual 
phases.

[discrete]
==== The phases of a {dfanalytics-job}

> A {dfanalytics-job} has two states: started and stopped. Until its state 

Reviewer 2: The states are actually many and in this logical order we have: starting, started, reindexing, analyzing, stopping, stopped, failed.

switches from started to stopped, the job goes through four phases regardless of 
the {dfanalytics} type it performs. The phases are discrete steps of the 
{dfanalytics-job} that is essentially a persistent {es} task. The four phases 
are the following:

* reindexing,
* loading data,
* analyzing,
* writing results.

Let's take a look at the phases one-by-one.

[discrete]
===== Reindexing

During the reindexing phase the documents from the source index or indices are 
copied to the destination index. If the destination index does not exist, then 
the job creates it before reindexing by using default settings. If it is already 
created, then the job will use the existing index. This provides flexibility for 
you to create the destination index and define any settings or mappings that you 
want to have in it. 

> If the destination index is built, the {dfanalytics-job} task calls the {es} 

Reviewer 2: Perhaps replace `If` with `Once`? The job will call the reindex API regardless of whether it had to create the destination index or not.

> {ref}/docs-reindex.html[Reindex API] to launch the reindexing task. When you 

Reviewer 2: I wonder if we need this last sentence. It seems like it's getting too low into implementation details.

Author: Yes, you are right. I felt that maybe it's too much detail.
